components/FetchedDogs/FetchedDogs.tsx

```diff
@@ -1,119 +1,86 @@
 import React, { useState, useEffect, useRef, useCallback } from "react";
-import { useLocalStorage } from "../../hooks/useLocalStorage";
+import { useFetchDog } from "../../hooks/useFetchDog";
 
-import {
-  DogDetails,
-  DogsDetails,
-  likedDogsDetails,
-  useAppContext,
-} from "../../context";
+import { likedDogsDetails, useAppContext } from "../../context";
 
 import { HeartButton } from "../Buttons/HeartButton/HeartButton";
 import { LoadingSpinner } from "../LoadingSpinner/LoadingSpinner";
 
 import style from "./FetchedDogs.module.css";
 import { Breed } from "../../data/breedsData";
-
-type fetchedDog = {
-  message: string;
-  status: string;
-};
+import Link from "next/link";
 
 export const FetchedDogs = () => {
   const {
+    isMosaic,
     loading,
     setLoading,
     fetchedDogs,
     setFetchedDogs,
     likedDogs,
     setLikedDogs,
     setIsError,
-    apiCallCounter,
-    setApiCallCounter,
+    storagedBreeds,
   } = useAppContext();
 
   const [breed, setBreed] = useState<Breed>();
   const [random, setRandom] = useState(0);
-  const [storagedBreeds] = useLocalStorage<Breed[]>("selectedBreeds", []);
-
+  const [isScreenFilled, setIsScreenFilled] = useState(false);
   const observer = useRef<null | IntersectionObserver>(null);
 
-  const getRandomBreed = () => {
+  const getNextDog = () => {
     setRandom(Math.floor(Math.random() * storagedBreeds.length));
 
     if (random === 0 || random === 1) {
       setRandom(-1);
     }
```

Comment:
Why set this state 2 times? Maybe just put this line above and the current 32 line in the else?